<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>HotkeyPage</class>
    <widget class="QWidget" name="HotkeyPage">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>800</width>
                <height>600</height>
            </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
            <property name="spacing">
                <number>20</number>
            </property>
            <property name="leftMargin">
                <number>40</number>
            </property>
            <property name="topMargin">
                <number>40</number>
            </property>
            <property name="rightMargin">
                <number>40</number>
            </property>
            <property name="bottomMargin">
                <number>40</number>
            </property>

            <item>
                <widget class="QLabel" name="label_title">
                    <property name="text">
                        <string>Hotkey Verwaltung</string>
                    </property>
                    <property name="class" stdset="0">
                        <string>page-title</string>
                    </property>
                </widget>
            </item>

            <!-- Configuration Card -->
            <item>
                <widget class="QGroupBox" name="group_config">
                    <property name="title">
                        <string>Konfiguration</string>
                    </property>
                    <layout class="QFormLayout" name="formLayout">
                        <item row="0" column="0">
                            <widget class="QLabel" name="label_activation">
                                <property name="text">
                                    <string>Aktivierung (Menü öffnen):</string>
                                </property>
                            </widget>
                        </item>
                        <item row="0" column="1">
                            <widget class="QComboBox" name="combo_activation">
                                <item>
                                    <property name="text">
                                        <string>Ctrl+Shift</string>
                                    </property>
                                </item>
                                <item>
                                    <property name="text">
                                        <string>Ctrl+Alt</string>
                                    </property>
                                </item>
                                <item>
                                    <property name="text">
                                        <string>Shift+Alt</string>
                                    </property>
                                </item>
                                <item>
                                    <property name="text">
                                        <string>Win+Shift</string>
                                    </property>
                                </item>
                            </widget>
                        </item>
                        <item row="1" column="0">
                            <widget class="QLabel" name="label_action">
                                <property name="text">
                                    <string>Aktions-Taste (gedrückt halten):</string>
                                </property>
                            </widget>
                        </item>
                        <item row="1" column="1">
                            <widget class="QComboBox" name="combo_action">
                                <item>
                                    <property name="text">
                                        <string>Alt</string>
                                    </property>
                                </item>
                                <item>
                                    <property name="text">
                                        <string>Ctrl</string>
                                    </property>
                                </item>
                                <item>
                                    <property name="text">
                                        <string>Shift</string>
                                    </property>
                                </item>
                                <item>
                                    <property name="text">
                                        <string>Win</string>
                                    </property>
                                </item>
                            </widget>
                        </item>
                        <item row="2" column="0">
                            <widget class="QLabel" name="label_duration">
                                <property name="text">
                                    <string>Verzögerung (Menü):</string>
                                </property>
                            </widget>
                        </item>
                        <item row="2" column="1">
                            <widget class="QDoubleSpinBox" name="spin_hold_duration">
                                <property name="minimum">
                                    <double>0.0</double>
                                </property>
                                <property name="maximum">
                                    <double>3.0</double>
                                </property>
                                <property name="singleStep">
                                    <double>0.1</double>
                                </property>
                                <property name="value">
                                    <double>0.5</double>
                                </property>
                                <property name="suffix">
                                    <string> s</string>
                                </property>
                            </widget>
                        </item>
                        <item row="3" column="1">
                            <widget class="QPushButton" name="btn_save_config">
                                <property name="text">
                                    <string>Speichern &amp; Neustarten</string>
                                </property>
                                <property name="maximumSize">
                                    <size>
                                        <width>200</width>
                                        <height>16777215</height>
                                    </size>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>

            <!-- Status Card -->
            <item>
                <widget class="QGroupBox" name="group_status">
                    <property name="title">
                        <string>Listener Status</string>
                    </property>
                    <layout class="QVBoxLayout" name="verticalLayout_status_card">
                        <item>
                            <layout class="QHBoxLayout" name="layout_status">
                                <item>
                                    <widget class="QLabel" name="lbl_status_icon">
                                        <property name="text">
                                            <string>⚪</string>
                                        </property>
                                        <property name="font">
                                            <font>
                                                <pointSize>24</pointSize>
                                            </font>
                                        </property>
                                        <property name="maximumSize">
                                            <size>
                                                <width>40</width>
                                                <height>16777215</height>
                                            </size>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="lbl_status_text">
                                        <property name="text">
                                            <string>Status unbekannt</string>
                                        </property>

                                    </widget>
                                </item>
                                <item>
                                    <spacer>
                                        <property name="orientation">
                                            <enum>Qt::Horizontal</enum>
                                        </property>
                                    </spacer>
                                </item>

                                <item>
                                    <widget class="QPushButton" name="btn_start">
                                        <property name="text">
                                            <string>Starten</string>
                                        </property>

                                    </widget>
                                </item>
                                <item>
                                    <widget class="QPushButton" name="btn_stop">
                                        <property name="text">
                                            <string>Stoppen</string>
                                        </property>

                                    </widget>
                                </item>
                                <item>
                                    <widget class="QPushButton" name="btn_restart">
                                        <property name="text">
                                            <string>Neustart</string>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>

                        <!-- NEW: Message Label -->
                        <item>
                            <widget class="QLabel" name="lbl_message">
                                <property name="text">
                                    <string></string>
                                </property>
                                <property name="alignment">
                                    <set>Qt::AlignCenter</set>
                                </property>

                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>

            <!-- Hotkey List -->
            <item>
                <widget class="QGroupBox" name="group_list">
                    <property name="title">
                        <string>Aktive Hotkeys (Alt + Taste)</string>
                    </property>
                    <layout class="QVBoxLayout" name="layout_list">
                        <item>
                            <widget class="QTableWidget" name="table_hotkeys">
                                <property name="columnCount">
                                    <number>2</number>
                                </property>
                                <property name="verticalScrollBarPolicy">
                                    <enum>Qt::ScrollBarAlwaysOff</enum>
                                </property>
                                <property name="horizontalScrollBarPolicy">
                                    <enum>Qt::ScrollBarAlwaysOff</enum>
                                </property>
                                <property name="sizePolicy">
                                    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                                        <horstretch>0</horstretch>
                                        <verstretch>0</verstretch>
                                    </sizepolicy>
                                </property>
                                <property name="selectionBehavior">
                                    <enum>QAbstractItemView::SelectRows</enum>
                                </property>
                                <property name="editTriggers">
                                    <enum>QAbstractItemView::NoEditTriggers</enum>
                                </property>

                                <column>
                                    <property name="text">
                                        <string>Taste</string>
                                    </property>
                                </column>
                                <column>
                                    <property name="text">
                                        <string>Aktion</string>
                                    </property>
                                </column>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>

        </layout>
    </widget>
</ui>
